crates/pathfinder/src/rpc/api.rs

```diff
@@ -505,11 +505,11 @@ impl RpcApi {
                     .map_err(internal_server_error)?;
 
                 return block
-                    .transactions
+                    .transactions()
                     .into_iter()
                     .nth(index)
                     .map_or(Err(ErrorCode::InvalidTransactionIndex.into()), |txn| {
-                        Ok(txn.into())
+                        Ok(txn.clone().into())
```

Comment:
Why would an clone be needed after into_iter + nth?

Reply:
The original type was a vector, whereas now its a slice. So that `into_iter` should actually be an `iter`.

Reply:
Yeah found out later. I think it's oki, just one of them. I don't think there's a good alternative, maybe to add more methods like `transactions_mut(&mut self)` to allow `std::mem::take` but ... seems tiresome.